### ISSUE TITLE
Techdocs won't refresh for entity definitions under non "master" branches in bitbucket

### ISSUE THREAD

**Issue by shanthisagar-dhanya-kumar-db** [backstage/backstage#16206]
### 📜 Description

Techdocs won't refresh for entity definitions under non "master" branches in bitbucket

### 👍 Expected behavior

commit tag of the branch used to register the component should be compared against the etag stored under techdocs-metadata.json

### 👎 Actual Behavior with Screenshots

Basically what is happening is the commit of the "master" branch is compared against the etag stored under techdocs-metadata.json which hasn't changed, hence no refreshes.

### 👟 Reproduction steps

We use bitbucket to store the entity yaml definitions.

Let's say we have an entity yaml definition in a feature branch, lets say feature/TEST-134. The very first time we register the component the techdocs gets generated. If there are any subsequent changes in any of the .md files or mkdocs.yml files on that specific branch, techdocs won't detect that to carry out refreshes.
Basically what is happening is the commit of the "master" branch is compared against the etag stored under techdocs-metadata.json which hasn't changed, hence no refreshes.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by sennyeya** [comment]
Can you explain a little bit more what you're trying to do? Ideally, documentation is stored on your primary branch as feature branches are transient. What's so special about `feature/TEST-1234` that it can't wait until `master` :P ?

Can you share a little more context on the yaml file that you're trying to create the techdocs from? At the very least, the `backstage.io/techdocs-ref` annotation. The `UrlReader` for techdocs should be able to read from non-master branches correctly and the `etag` is being returned from the upstream response.
